pipelines/curate_coding.py

```
@@ -525,6 +586,584 @@ def curate_jsonl(
+            violations.append(mismatch)
+
+        counts = manifest.get("step_counts")
+        actions = manifest.get("step_actions")
+        if not isinstance(counts, dict) or not isinstance(actions, list):
+            violations.append(f"{where}: manifest records no step accounting")
+            continue
+        if PRE_STEP_EXCLUSION_REASONS.intersection(reasons):
+            if actions or any(
+                counts.get(key) not in (0, None)
+                for key in ("source", "retained", "migrated", "excluded")
+            ):
```

Ruff (0.16.2)

[warning] Use a single `if` statement instead of nested `if` statements

(SIM102)